sonic_installer/bootloader/aboot.py

```diff
@@ -164,7 +165,22 @@ def get_binary_image_version(self, image_path):
         return IMAGE_PREFIX + version.strip()
 
     def verify_image_platform(self, image_path):
-        return os.path.isfile(image_path)
+        if not os.path.isfile(image_path):
+            return False
+
+        # Get running platform
+        platform = device_info.get_platform()
+
+        # If .platforms_asic is not existed, we simply return True for backward
+        # compatibility. Otherwise, we check if current platform is inside the
+        # supported target platforms list.
+        try:
+            output = subprocess.check_output(['/usr/bin/unzip', '-qop', image_path, '.platforms_asic'], text=True)
+            return platform in output
+        except subprocess.CalledProcessError:
```

Comment:
>CalledProcessError [](http://example.com/codeflow?start=26&length=18)

This section is good consideration for downgrade use case.

However, `CalledProcessError` is larger than the the case ".platforms_asic is not existed". Could you check it in straighforward way?

Reply:
I am refering here. Seems the image_path is guaranteed to exist. So it only raise exeption if platform_asic is not existed. Maybe it is enough to just for checking existence?
[image_version](https://github.com/Azure/sonic-utilities/blob/master/sonic_installer/bootloader/aboot.py#L159)